oak-core/src/test/java/org/apache/jackrabbit/oak/query/QueryInformationLoggerTest.java

```diff
@@ -0,0 +1,143 @@
+            internal.starting();
+            String query = "SELECT * FROM [cq:Page]";
+            QueryInformationLogger.logCaller(query, ignoredClasses);
+            assertEquals("Exactly 1 entry must be written",1,external.getLogs().size());
+            String logEntry = external.getLogs().get(0);
+            assertTrue(logEntry.contains("org.junit.runners.model")); // org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
+            assertEquals(0,internal.getLogs().size());
+        } finally {
+            external.finished();
+            internal.finished();
+        }      
+    }
+    
+    @Test
+    public void withStacktrace() {
+        
+        LogCustomizer external = LogCustomizer.forLogger(EXTERNAL_LOG).enable(Level.DEBUG).create();
+        LogCustomizer internal = LogCustomizer.forLogger(INTERNAL_LOG).enable(Level.TRACE).create();
+        try {
+            external.starting();
+            internal.starting();
+            String query = "SELECT * FROM [cq:Page]";
+            QueryInformationLogger.logCaller(query, ignoredClasses);
+            assertEquals("Exactly 2 entries must be written",2,external.getLogs().size()); //     
+            assertEquals(0,internal.getLogs().size());
+        } finally {
+            external.finished();
+            internal.finished();
+        }      
+    }
+    
+    @Test
+    public void testOakInternalStatement() {
+        LogCustomizer external = LogCustomizer.forLogger(EXTERNAL_LOG).enable(Level.DEBUG).create();
+        LogCustomizer internal = LogCustomizer.forLogger(INTERNAL_LOG).enable(Level.INFO).create();
+        try {
+            external.starting();
+            internal.starting();
+            String query = "SELECT * FROM [cq:Page] " + QueryInformationLogger.OAK_INTERNAL_MARKER;
+            QueryInformationLogger.logCaller(query, ignoredClasses);
+            
+            // On INFO no logs are written
+            assertEquals(0,external.getLogs().size()); //     
+            assertEquals(0,internal.getLogs().size());
+        } finally {
+            external.finished();
+            internal.finished();
+        }     
+    }
+    
+    @Test
+    public void testWithAllStackEntriesIgnored_INFO() {
+        LogCustomizer external = LogCustomizer.forLogger(EXTERNAL_LOG).enable(Level.DEBUG).create();
+        LogCustomizer internal = LogCustomizer.forLogger(INTERNAL_LOG).enable(Level.INFO).create();
+        try {
+            external.starting();
+            internal.starting();
+            String query = "SELECT * FROM [cq:Page]";
+            QueryInformationLogger.logCaller(query, allClassesIgnored);
+            try {
+                throw new AssertionError();
+            } catch (AssertionError e) {
+                e.printStackTrace();
+            }
+            
+            assertEquals(0,external.getLogs().size());    
+            assertEquals(0,internal.getLogs().size()); // would require DEBUG
```

Comment:
- formatting again :)
- you could simplify as assertTrue(external.getLogs().isEmpty().. but that's probably a matter of taste.